lib/render.js

```diff
@@ -1432,3 +1432,25 @@ function getNodeAtPath(el, path) {
   }
   return el
 }
+
+/**
+ * Retrieve the nearest 'body' ancestor of the given element or else the root
+ * element of the document in which stands the given element.
+ *
+ * This is necessary if you want to attach the events handler to the correct
+ * element and be able to dispatch events in document fragments such as
+ * Shadow DOM.
+ *
+ * @param  {HTMLElement} el The element on which we will render an app.
+ * @return {HTMLElement}    The root element on which we will attach the events
+ *                          handler.
+ */
+function getRootElement(el) {
+  while (true) {
+    if (el.tagName === 'BODY' || !el.parentElement) {
+      return el;
+    }
+
+    el = el.parentElement;
+  };
```

Comment:
Not with the `while (true)` :)
But I can see why you wouldn't want a `while (true)` in the code. This is rarely synonym of good fortune.